site/app/controllers/student/GradeInquiryController.php

```diff
@@ -112,7 +112,11 @@ public function makeGradeInquiryPost($gradeable_id) {
 
         try {
             $regrade_post_id = $this->core->getQueries()->insertNewRegradePost($grade_inquiry_id, $user->getId(), $content, $gc_id);
-            $regrade_post = $this->core->getQueries()->getRegradePost($regrade_post_id);
+            $regrade_post = $this->core->getQueries()->getRegradePost($regrade_post_id, $grade_inquiry_id);
+            if ($regrade_post === null) {
```

Comment:
How does `$regrade_post` be null here? It seems to me that `getRegradePost($regrade_post_id, $grade_inquiry_id)` is just going to get always fetch the post that was just inserted right above as part of `insertNewRegradePost`?